### ISSUE TITLE
🚀 Feature: Parallel support for test (it()) block

### ISSUE THREAD

**Issue by rakeshpareek01** [mochajs/mocha#4983]
**Is your feature request related to a problem or a nice-to-have?? Please describe.**
Running the test suite in parallel creates conflict when Test Cases in a suite (describe) block are dependent on any database configuration and the same configuration is changing at runtime in other different file test suites as per the Test Cases in `describe` block
In this use-case, every test suite has a before and after block to set some settings on an application, and in the same project every test suite uses the same application but on a different set of settings (set in before) so running describe block parallel is not a good idea as `before` block from multiple suites will run parallel and that will create conflict and tests will fail.

**Describe the solution you'd like**
There should be an option to run Test Cases (all `it` blocks) in Parallel inside of a describe block similar to `mocha.parallel`. I have tried using mocha.parallel but it has a lot of limitations like it does not support mocha config, reties etc...

**Describe alternatives you've considered**
`mocha.parallel` is a good option but it has a lot of limitations for mocha features

**Comment by JoshuaKGoldberg** [comment]
What you're describing basically _is_ parallel mode! 😄

We're going to want to add any major new parallel-equivalent modes. Especially given #5027, we just don't have the budget. And even if we did, I think a better path forward would be to make parallel mode able to satisfy your use case. 

> limitations

Let's talk about this more. What specifically are the limitations you're bumping into? 

> does not support mocha config

Err, doesn't it? It works on my machine. Nothing in the docs indicates it wouldn't. Could you please provide a full reproduction showing it not?

> retries

Same question 🙂 .

**Comment by JoshuaKGoldberg** [comment]
Closing out old issues. If someone does have something more to add, please do post back - we'd be happy to take a look! 🤎
